tests/functional/a/abstract/abstract_class_instantiated.py

```diff
@@ -141,3 +143,61 @@ def test(self):
 def main_two():
     """ do nothing """
     BadClassTwo() # [abstract-class-instantiated]
+
+
+# Testcase from https://github.com/PyCQA/pylint/issues/3060
+with pd.ExcelWriter("demo.xlsx") as writer:
+    print(writer)
+
+class GoodWithNew(metaclass=abc.ABCMeta):
+    def __new__(cls):
+        pass
+
+    @property
+    @abc.abstractmethod
+    def sheets(self):
+        """sheets."""
+
+    @property
+    @abc.abstractmethod
+    def book(self):
+        """book."""
+
+test = GoodWithNew()
+
+
+class BadWithNew(metaclass=abc.ABCMeta):
+    def __new__(cls):
+        print("Test.__new__")
+        return super().__new__(cls)
+
+    @property
+    @abc.abstractmethod
+    def sheets(self):
+        """sheets."""
+
+    @property
+    @abc.abstractmethod
+    def book(self):
+        """book."""
+
+test = BadWithNew()  # [abstract-class-instantiated]
+
+
+class GoodithComplexNew(metaclass=abc.ABCMeta):
+    def __new__(cls):
+        warnings.warn("You are calling super on an abstract class.")
+        return object.__new__(cls)
+
+    @property
+    @abc.abstractmethod
+    def sheets(self):
+        """sheets."""
+
+    @property
+    @abc.abstractmethod
+    def book(self):
+        """book."""
+
+
+bad = GoodithComplexNew()
```

Comment:
What about:

```python
def __new__(self):
    cls = super().__new__()
    return cls

def __new__(self):
    cls = super().__new__()
    return "1"
```

Reply:
as long as the class has abstract methods, yes msg will be emitted. will add them as test cases